### ISSUE TITLE
Web chat UI improvement

### ISSUE THREAD

**Issue by D4nte** [status-im/js-waku#51]
Nice to have improvements:
- Auto scroll when messages overflow from window
- Dynamic size of chat room to fill the screen without expending when too much text shows
- ~Display self message differently (bold?)~
- ~Allow configuration of nick via UI~
- ~Show command response in differently (yellow background?)~

**Comment by shivekkhurana** [comment]
### Dynamic size of chat room to fill the screen without expending when too much text shows
One way to go about this is to have a flex-box with two elements : `chat-list` and `chat-input`. We can make the `chat-list` grow to occupy all space. This will put the chat input to the bottom.

Here's an example: https://jsbin.com/qiwayequto/edit?html,css,output

**Comment by shivekkhurana** [comment]
### Auto scroll when messages overflow from window
For this we can implement a `useEffect` hook in the `Lines` component in `Room.tsx`. The idea is to make the window scroll to bottom when `messages` prop changes, or each time the component re-renders. 

I found this clever example on StackOverflow (check the topmost answer): https://stackoverflow.com/questions/45719909/scroll-to-bottom-of-an-overflowing-div-in-react/45837342

To make it more robust, we can modify the `useEffect` hook to trigger only when the `messages` prop changes:

```js
const AlwaysScrollToBottom = ({messages}) => {
  const elementRef = useRef();
  useEffect(() => elementRef.current.scrollIntoView(), [messages]); // <-------- Added messages as a dependancy
  return <div ref={elementRef} />;
};

const Messages = ({ items }) => (
  <ul>
    {items.map(({id, text}) => <li key={id}>{text}</li>)}
    <AlwaysScrollToBottom messages={items}/> // <-------- Pass list items as messages
  </ul>
)

const App = () => (
  <div className="container">
    <Messages items={[...]}>
    <MessageComposer />
  </div>
)

```

**Comment by D4nte** [comment]
> ### Dynamic size of chat room to fill the screen without expending when too much text shows
> 
> One way to go about this is to have a flex-box with two elements : `chat-list` and `chat-input`. We can make the `chat-list` grow to occupy all space. This will put the chat input to the bottom.
> 
> Here's an example: [jsbin.com/qiwayequto/edit?html,css,output](https://jsbin.com/qiwayequto/edit?html,css,output)
> and the same example with scroll in place: [jsbin.com/xazatopepu/1/edit?html,css,output](https://jsbin.com/xazatopepu/1/edit?html,css,output)

Thanks! It looks much better. I did it in bc3328463ba8799c572f728904eea9815f96679b
 However, when the text reaches the full height, the input box slowly starts to go down an a second scroll bar appears on the right. See screenshot. Is there a way to avoid that? Or is my implementation incorrect?

![Screenshot from 2021-04-26 17-33-24](https://user-images.githubusercontent.com/300805/116045826-c6d3e900-a6b5-11eb-8646-b22a78d6125e.png)

**Comment by shivekkhurana** [comment]
This extra scroll indicates that some element (or the browser) is adding extra padding/ margin. You should include a css reset in your app [1][2].

If reset doesn't work, you can force the height and width of the parent to be 100vh and 100vw respectively. And set overflow to hidden. 

[1] HIstory of reset.css <https://meyerweb.com/eric/tools/css/reset/>
[2] Adding Reset CSS to CRA projects <https://create-react-app.dev/docs/adding-css-reset/>

**Comment by D4nte** [comment]
Hum. I tried both
- https://create-react-app.dev/docs/adding-css-reset/
- AND https://github.com/csstools/normalize.css#browser-support

But no luck. I then tried the parent fix you suggested. the double scrollbar have now disappeared but the input box still slowly go down as new messages appear. Once it reaches the "bottom" after few messages it stays there.
See commit 9a028e793c472b62e25ed20c278b791345773367
Any idea how to fix this?
